### ISSUE TITLE
cpu_affinity() segfaults on Linux on CentOS / manylinux

### ISSUE THREAD

**Issue by giampaolo** [giampaolo/psutil#1126]
See: https://github.com/giampaolo/psutil/issues/824#issuecomment-328330124

```
psutil.tests.test_process.TestProcess.test_cmdline ... ok
psutil.tests.test_process.TestProcess.test_cpu_affinity ... ok
psutil.tests.test_process.TestProcess.test_cpu_affinity_errs ... config.sh: line 17:    55 Segmentation fault      (core dumped) PSUTIL_TESTING=1 python -Wa ../psutil/psutil/tests/__main__.py
```

**Comment by giampaolo** [comment]
Note: as far as I understand the segfault occurs when using the wheel built on CentOS 5. That means a different implementation of `psutil_proc_cpu_affinity_get` is used because `CPU_ALLOC` is not defined: 
https://github.com/giampaolo/psutil/blob/fbece8e45991e8cc070bfd5e9efcd4974227abcd/psutil/_psutil_linux.c#L350

**Comment by webknjaz** [comment]
FTR I've faced a similar issue when building `manylinux1` wheel for pygit2 (https://github.com/libgit2/pygit2/issues/793#issuecomment-464371085): tests segfaulted under the same manylinux1 container. But the wheels themselves worked under other envs + they worked after changing directory on my dev machine and crashed when I tried using those from within project dir probably picking up *something* from a current working directory rather than virtualenv...

**Comment by giampaolo** [comment]
To further elaborate on this, and as a note to self: 
* the change was introduced because of https://github.com/giampaolo/psutil/issues/536
* and merged here: https://github.com/giampaolo/psutil/pull/537
* the [alternate method](https://github.com/giampaolo/psutil/blob/fbece8e45991e8cc070bfd5e9efcd4974227abcd/psutil/_psutil_linux.c#L350) should be tested/fixed on CentOS

**Comment by webknjaz** [comment]
@giampaolo is this still a problem with wheels shipped as v0.28.0?
